lightning/src/ln/router.rs

```diff
@@ -1049,6 +1085,28 @@ impl Router {
 
 		Err(LightningError{err: "Failed to find a path to the given destination", action: ErrorAction::IgnoreError})
 	}
+
+	/// Gets the list of announced channels, in random order. See ChannelInfo details field documentation for more information
+	pub fn list_edges(&self) -> Vec<ChannelInfo> {
+		let mut edges = Vec::new();
+		let network = self.network_map.read().unwrap();
+		edges.reserve(network.channels.len());
+		for (_, chan) in network.channels.iter() {
+			edges.push((*chan).clone());
```

Comment:
The whole point in using an internal representation was to avoid copying the whole thing. Just return network.channels.iter() itself. (this also means you dont have to include the short_channel_ids duplicatively in memory).

Reply:
But you can't return a reference to network here as it's under a lock and you can't sure than lock taking is going to be successful?

Reply:
Grrrr, I forgot you can't return the lock and an interator from inside it :/. Can you remove the commit that adds the short_channel_id and just return (u64, ChannelInfo)?